xandikos/web.py

```diff
@@ -1030,6 +1030,14 @@ def get_resource(self, relpath):
             except KeyError:
                 return None
 
+    def set_principal(self, user):
+        principal = "/%s/" % user
```

Comment:
Ideally this path would be configurable, but we could leave that for a follow-up PR.

Reply:
Do you mean something like:
``` python
def set_principal(self, user, principal_path_prefix="/", principal_path_suffix="/"):
    principal = principal_path_prefix + user + principal_path_suffix
```

Reply:
Yep, exactly. Though we'd probably want those extra paths to come from a command-line argument/configuration file.

Reply:
Can you explain SCRIPT_NAME?

Reply:
SCRIPT_NAME is a bit of a misnomer, but it's named after a similar variable in the CGI world - it's the bit of the path to Xandikos itself. If Xandikos is under https://foo.example.com/dav/, then SCRIPT_NAME would be /dav/.